frontend/src/lib/components/SaveToDashboard/saveToDashboardModalLogic.ts

```diff
@@ -47,9 +47,37 @@ export const saveToDashboardModalLogic = kea<saveToDashboardModalLogicType<SaveT
         _dashboardId: [null as null | number, { setDashboardId: (_, { id }) => id }],
         searchQuery: ['', { setSearchQuery: (_, { query }) => query }],
         scrollIndex: [-1 as number, { setScrollIndex: (_, { index }) => index }],
+        adding: [false, { addToDashboard: () => true, reportSavedInsightToDashboard: () => false }],
+        removing: [
+            false,
+            {
+                removeFromDashboard: () => true,
+                reportRemovedInsightFromDashboard: () => false,
```

Comment:
I don't know why, but this feels dirty somehow -> depending on some auxiliary reporting action to fire, and updating an internal state because of that. But I'm too tired to nit any more 🙃

Reply:
💡